src/ioctls/vm.rs

```diff
@@ -1279,12 +1279,57 @@ mod tests {
             assert!(vm_fd.register_irqfd(&evtfd2, 8).is_ok());
             assert!(vm_fd.register_irqfd(&evtfd3, 4).is_ok());
         }
-
         // On aarch64, this fails because setting up the interrupt controller is mandatory before
         // registering any IRQ.
         // On x86_64 this fails as the event fd was already matched with a GSI.
         assert!(vm_fd.register_irqfd(&evtfd3, 4).is_err());
         assert!(vm_fd.register_irqfd(&evtfd3, 5).is_err());
+
```

Comment:
I think we are better of with 2 separate tests for x86_64 and for aarch64. The code is widely different depending on the platform. Maybe we can also have register and unregister tested in the same test?

Reply:
Hi @andreeaflorescu ! I think it perfectly makes sense, and therefore I have made some code changes:

- Seperate the x86 test and the aarch64 test into two test functions.

- Rename the test function as `test_register_unregister_irqfd` and test both register and unregister irqfd in the body.

Please review.